plugins/pipelines/create.js

```diff
@@ -48,11 +48,17 @@ module.exports = () => ({
             const userFactory = request.server.app.userFactory;
             const scmUrl = formatScmUrl(request.payload.scmUrl);
             const username = request.auth.credentials.username;
+            let user;
 
             // fetch the user
             return userFactory.get({ username })
                 // get the user permissions for the repo
-                .then(user => user.getPermissions(scmUrl))
+                .then(u => {
+                    // store the user object for fetching token later
+                    user = u;
```

Comment:
You don't need to do this. You can have the rest nested inside `.then(user =>` scope and `user` will be available.